sympy/series/sequences.py

```diff
@@ -0,0 +1,1115 @@
+        sequences if they can simplify themselves with any other constituent
+
+        Notes
+        =====
+        adapted from ``Union.reduce``
+
+        """
+        new_args = True
+        while(new_args):
+            for id1, s in enumerate(args):
+                new_args = False
+                for id2, t in enumerate(args):
+                    if id1 == id2:
+                        continue
+                    new_seq = s._add(t)
+                    # This returns None if s does not know how to add
+                    # with t. Returns the newly added sequence otherwise
+                    if new_seq is not None:
+                        new_args = [a for a in args if a not in (s, t)]
+                        new_args.append(new_seq)
+                        break
+                if new_args:
+                    args = new_args
+                    break
+
+        if len(args) == 1:
+            return args.pop()
+        else:
+            return SeqAdd(args, evaluate=False)
+
+    def _eval_coeff(self, pt):
+        """adds up the coefficients of all the sequences at point pt"""
+        return sum(a.coeff(pt) for a in self.args)
+
+
+class SeqMul(Mul, SeqExprOp):
+    """
+    Represents term-wise multiplication of sequences.
+    Handles multiplication of sequences only. For multiplication
+    with other objects see ``SeqBase.coeff_mul``.
+
+    Rules:
+        * The interval on which sequence is defined is the intersection
+        of respective intervals of sequences.
+        * Anything * EmptySequence returns EmptySequence
+        * Other rules are defined in _mul methods of sequence classes
+
+    Examples
+    ========
+
+    >>> from sympy import S, oo, SeqMul, SeqPer, SeqFormula
+    >>> from sympy.abc import n
+    >>> SeqMul(SeqPer((1, 2)), S.EmptySequence)
+    EmptySequence()
+    >>> SeqMul(SeqPer((1, 2), (0, 5)), SeqPer((1, 2), (6, 10)))
+    EmptySequence()
+    >>> SeqMul(SeqPer((1, 2)), SeqFormula(n**2))
+    SeqPer((1, 2), ([0, oo), 1))*SeqFormula((n**2, n), ([0, oo), 1))
+    >>> SeqMul(SeqFormula(n**3), SeqFormula(n**2))
+    SeqFormula((n**5, n), ([0, oo), 1))
+
+    See Also
+    ========
+
+    sympy.series.sequences.SeqAdd
+    """
+
+    def __new__(cls, *args, **kwargs):
+        evaluate = kwargs.get('evaluate', global_evaluate[0])
+
+        # flatten inputs
+        args = list(args)
+
+        # adapted from sympy.sets.sets.Union
+        def _flatten(arg):
+            if isinstance(arg, SeqBase):
+                if isinstance(arg, SeqMul):
+                    return sum(map(_flatten, arg.args), [])
+                else:
+                    return [arg]
+            elif iterable(arg):
+                return sum(map(_flatten, arg), [])
+            raise TypeError("Input must be Sequences or "
+                            " iterables of Sequences")
+        args = _flatten(args)
+
+        # Multiplication of no sequences is EmptySequence
+        if len(args) == 0:
```

Comment:
I think this will be confusing. args is a list. Now first impression of ``if not args`` is  something like a variable. I think len(args) serves a better purpose. Lot clearer to me.